fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -518,6 +531,16 @@ private Loan(final String accountNo, final Client client, final Group group, fin
         // rates added here
         this.rates = rates;
 
+        if (loanProduct.isRevolving()) {
+            if (revolvingPeriodStartDate != null) {
+                this.revolvingPeriodStartDate = Date
+                        .from(revolvingPeriodStartDate.atStartOfDay(DateUtils.getDateTimeZoneOfTenant()).toInstant());
+                if (revolvingPeriodEndDate != null) {
+                    this.revolvingPeriodEndDate = Date
+                            .from(revolvingPeriodEndDate.atStartOfDay(DateUtils.getDateTimeZoneOfTenant()).toInstant());
```

Comment:
change to `endOfDay`

Reply:
there is no such method... regardless, this should be used without regard to the time.. I think...

Reply:
why accounting is related? 
any why, in the current implementation you cant do a disbursement in the end date which is wrong.

https://stackoverflow.com/questions/36408548/how-to-obtain-the-end-of-the-day-when-given-a-localdate